airflow/models/dag.py

```diff
@@ -522,17 +522,35 @@ def __init__(
         self.has_on_success_callback = self.on_success_callback is not None
         self.has_on_failure_callback = self.on_failure_callback is not None
 
-        self.doc_md = doc_md
-
         self._access_control = DAG._upgrade_outdated_dag_access_control(access_control)
         self.is_paused_upon_creation = is_paused_upon_creation
 
         self.jinja_environment_kwargs = jinja_environment_kwargs
         self.render_template_as_native_obj = render_template_as_native_obj
+
+        self.doc_md = self.get_doc_md(doc_md)
+
         self.tags = tags or []
         self._task_group = TaskGroup.create_root(self)
         self.validate_schedule_and_params()
 
+    def get_doc_md(self, doc_md: str) -> str:
+        if doc_md is None:
+            return doc_md
+
+        env = self.get_template_env(force_sandboxed=False)
+
+        if not doc_md.endswith('.md'):
```

Comment:
```suggestion
        if not doc_md.endswith('.md'):
```

I think we also nee to handle the case (and add tests). where the file is missing (and treat it is document). I can easily imagine a case when somoene puts a description:

"And then you might also see more in this fancy document.md") 

Which would trigger a not-nice exception.

Reply:
it will trigger a `jinja2.exceptions.TemplateNotFound:` from Jinja followed by the name of the file :)

As you can see [here](https://github.com/apache/airflow/runs/7672321434?check_suite_focus=true#step:10:10794)

Reply:
Yep. Not nice if yoy really wanted your documentation end with ".md" :D. I think I'd prefer in this case to continue running but simply to use the "your documentation,md" as the content of the doc. I don't think failing DAG parsing in this case is justified.

Reply:
I trying to trigger a similar error with the PG operator with a query ending with `.sql`
<img width="694" alt="image" src="https://user-images.githubusercontent.com/3058143/183014884-85793dd2-b734-4d57-9f96-7ab9f6558680.png">

Would it make sense to return an error message in the doc, something like:

doc_md = f""""
# Templating Error!
Not able to find the template file: {doc_md}.
"""